lib/auth/integration/integrationv1/awsoidc.go

```diff
@@ -897,3 +898,75 @@ func (s *AWSOIDCService) Ping(ctx context.Context, req *integrationpb.PingReques
 		UserId:    resp.UserID,
 	}, nil
 }
+
+func (s *Service) deleteAWSOIDCAssociatedResources(ctx context.Context, authCtx *authz.Context, ig types.Integration) error {
+	if err := authCtx.CheckAccessToKind(types.KindDiscoveryConfig,
+		types.VerbList, types.VerbUpdate, types.VerbDelete); err != nil {
+		return trace.AccessDenied(
+			`deleting AWS OIDC integration requires 'list', 'update', and 'delete' ` +
+				`permissions for 'discovery_config' kind to remove associated resources`)
+	}
+
+	if err := authCtx.CheckAccessToKind(types.KindAppServer,
+		types.VerbList, types.VerbDelete); err != nil {
+		return trace.AccessDenied(
+			`deleting AWS OIDC integration requires 'list' and 'delete' ` +
+				`permissions for 'app_server' kind to remove associated resources`)
+	}
```

Comment:
All the other methods here check access to `KindIntegration` which I think we should follow here as well. It would be weird if creating the integration requires you to have access to one resource and deleting it to another. Since this only removes resources specific to this integration, it should be ok to check against integration kind.

Reply:
Hmm I see. In that case it is probably fine, and yeah, a comment would be helpful. Do we require permissions to create/update KindDiscoveryConfig and KindAppServer when creating the integration? @marcoandredinis

Reply:
We don't require those permissions when creating the integration.
They are only checked when using the flows:
- `discovery_config.<verb>` when setting up auto discover
- `app_server.<verb>` when enabling AWS App Access